modules/programs/waybar.nix

```diff
@@ -0,0 +1,378 @@
+          example = literalExample ''
+            [ "sway/workspaces" "sway/mode" "wlr/taskbar" ]
+          '';
+        };
+
+        modules-center = lib.mkOption {
+          type = nullOr (listOf str);
+          default = null;
+          description = "Modules that will be displayed in the center.";
+          example = literalExample ''
+            [ "sway/window" ]
+          '';
+        };
+
+        modules-right = lib.mkOption {
+          type = nullOr (listOf str);
+          default = null;
+          description = "Modules that will be displayed on the right.";
+          example = literalExample ''
+            [ "mpd" "custom/mymodule#with-css-id" "temperature" ]
+          '';
+        };
+
+        modules = lib.mkOption {
+          type = attrsOf unspecified;
+          default = { };
+          description = "Modules configuration.";
+          example = literalExample ''
+            {
+              "sway/window" = {
+                max-length = 50;
+              };
+              "clock" = {
+                format-alt = "{:%a, %d. %b  %H:%M}";
+              };
+              "custom/hello-from" = {
+                format = "hello {}";
+                max-length = 40;
+                interval = 10;
+                # If you have multiple scripts defined inline, you may be interested in
+                # using symlinkJoin to merge all scripts in one derivation
+                # to have them all under one directory structure in the nix store
+                exec = "''${pkgs.writers.writeBashBin "hello-from-waybar" '''
+                  echo "from within waybar"
+                '''}/bin/hello-from-waybar";
+              };
+            }
+          '';
+        };
+
+        margin = lib.mkOption {
+          type = nullOr str;
+          default = null;
+          description = "Margins value using the CSS format without units.";
+          example = "20 5";
+        };
+
+        inherit (margins) margin-top margin-left margin-bottom margin-right;
+
+        name = lib.mkOption {
+          type = nullOr str;
+          default = null;
+          description =
+            "Optional name added as a CSS class, for styling multiple waybars.";
+          example = "waybar-1";
+        };
+
+        gtk-layer-shell = lib.mkOption {
+          type = nullOr bool;
+          default = null;
+          example = false;
+          description =
+            "Option to disable the use of gtk-layer-shell for popups.";
+        };
+      };
+    };
+in {
+  meta.maintainers = [ lib.hm.maintainers.berbiche ];
+
+  options.programs.waybar = with lib.types; {
+    enable = lib.mkEnableOption "Waybar";
+
+    package = lib.mkOption {
+      type = nullOr package;
+      default = pkgs.waybar;
+      defaultText = literalExample "${pkgs.waybar}";
+      description = ''
+        Waybar package to use. Set to <code>null</code> to use the default module.
```

Comment:
This seems incorrect. Won't, e.g., `ExecStart = "${cfg.package}/bin/waybar";` fail if you try to assign `null` to this option? As far as I know all other `package` options simply have type `types.package` so perhaps it could be the same here?

Reply:
I'm not sure how merging works when both `nullOr` and default are present. I assumed `nullOr` was required for `default` to take place (which is not the case).